lemur/plugins/lemur_azure/plugin.py

```diff
@@ -255,13 +177,103 @@ class AzureSourcePlugin(SourcePlugin):
     author = "Bob Shannon"
     author_url = "https://github.com/datadog/lemur"
 
+    options = [
+        {
+            "name": "authenticationMethod",
+            "type": "select",
+            "value": "azureApp",
+            "required": True,
+            "available": ["hashicorpVault", "azureApp"],
+            "helpMessage": "Authentication method to use",
+        },
+        {
+            "name": "azureTenant",
+            "type": "str",
+            "required": True,
+            "validation": check_validation("^([a-zA-Z0-9-?])+$"),
+            "helpMessage": "Tenant to discover Azure resources from.",
+        },
+        {
+            "name": "azureAppID",
+            "type": "str",
+            "required": False,
+            "validation": check_validation("^([a-zA-Z0-9-?]?)+$"),
+            "helpMessage": "AppID for the Azure Key Vault. Required if authentication method is 'azureApp'.",
+        },
+        {
+            "name": "azurePassword",
+            "type": "str",
+            "required": False,
+            "validation": check_validation("([0-9a-zA-Z.:_-~]?)+"),
+            "helpMessage": "Tenant password for the Azure Key Vault. Required if authentication method is 'azureApp'.",
+        },
+        {
+            "name": "hashicorpVaultMountPoint",
+            "type": "str",
+            "required": False,
+            "helpMessage": "Path the Azure secrets engine was mounted on. Required if authentication "
+                           "method is 'hashicorpVault'.",
+        },
+        {
+            "name": "hashicorpVaultRoleName",
+            "type": "str",
+            "required": False,
+            "helpMessage": "Name of the role to fetch credentials for. Required if authentication "
+                           "method is 'hashicorpVault'.",
+        },
+    ]
+
     def __init__(self, *args, **kwargs):
         super(AzureSourcePlugin, self).__init__(*args, **kwargs)
 
     def get_certificates(self, options, **kwargs):
         # TODO(EDGE-1725) Support discovering endpoints and certificates in Azure source plugin
-        pass
+        return []
 
-    def get_endpoints(self, options, **kwargs):
+    def get_certificate_by_name(self, certificate_name, options):
         # TODO(EDGE-1725) Support discovering endpoints and certificates in Azure source plugin
-        pass
+        return
+
+    def get_endpoints(self, options, **kwargs):
+        credential = get_azure_credential(self, options)
+
+        endpoints = []
+        for subscription in SubscriptionClient(credential=credential).subscriptions.list():
```

Comment:
Our Azure deployments have a lot of subscriptions which could change over time for a single tenant as k8s clusters churn so such a change could potentially introduce a lot of configuration overhead.

For rotation, I think we'll just have the destination plugin upload the certificate to the key vault and rely on Azure to perform the rotation. If we really wanted to rotate the endpoint in the source plugin we could probably just use the full resource ID of the app gateway (which includes the subscription ID) as the endpoint name.

Reply:
> Our Azure deployments have a lot of subscriptions which could change over time for a single tenant as k8s clusters churn so such a change could potentially introduce a lot of configuration overhead.

gotcha, I thought the subscriptions were static. Given that this isn't likely to happen since each tenant represents a site I'm fine with keeping source == Azure tenant.